src/Mobs/Path.cpp

```diff
@@ -113,15 +109,24 @@ ePathFinderStatus cPath::Step()
 
 
 
-#ifndef COMPILING_PATHFIND_DEBUGGER
 bool cPath::IsSolid(const Vector3d & a_Location)
 {
-	int ChunkX, ChunkZ;
-	m_Item_CurrentBlock = a_Location;
-	cChunkDef::BlockToChunk(a_Location.x, a_Location.z, ChunkX, ChunkZ);
-	return !m_World->DoWithChunk(ChunkX, ChunkZ, * this);
+	m_Chunk = m_Chunk->GetNeighborChunk(a_Location.x, a_Location.z);
+	if (!m_Chunk->IsValid())
+	{
+		return true;
+	}
+	BLOCKTYPE BlockType;
+	NIBBLETYPE BlockMeta;
+	int RelX = a_Location.x - m_Chunk->GetPosX() * cChunkDef::Width;
+	int RelZ = a_Location.z - m_Chunk->GetPosZ() * cChunkDef::Width;
+	m_Chunk->GetBlockTypeMeta(RelX, a_Location.y, RelZ, BlockType, BlockMeta);
+	if ((BlockType == E_BLOCK_FENCE) || (BlockType == E_BLOCK_FENCE_GATE))
```

Comment:
"Rather than depending on "someone else" to set the solidity for you, how about checking the block below yourself instead?", I just wanted to squeeze some performance by not calling GetCell unless really needed. You may think I'm micro-optimizing, but this function is the most-called function in the PathFinder (and possibly in the whole game, given enough mobs).